common/persistence/sql/sqlExecutionStore.go

```diff
@@ -252,119 +254,119 @@ func (m *sqlExecutionStore) GetWorkflowExecution(
+
+	g, ctx := errgroup.WithContext(ctx)
+
+	g.Go(func() (e error) {
+		defer recoverPanic(&e)
+		executions, executionsError = m.getExecutions(ctx, request, domainID, wfID, runID)
+		// try catching executions error if we can but it's not guaranteed
+		// if one of the other futures return before here we won't capture the executions error
+		e = executionsError
+		return e
+	})
 
-	{
-		var err error
-		state.ChildExecutionInfos, err = getChildExecutionInfoMap(
-			ctx,
-			m.db,
-			m.shardID,
-			domainID,
-			wfID,
-			runID,
-			m.parser)
-		if err != nil {
-			return nil, err
-		}
-	}
+	g.Go(func() (e error) {
+		defer recoverPanic(&e)
+		activityInfos, e = getActivityInfoMap(
+			ctx, m.db, m.shardID, domainID, wfID, runID, m.parser)
+		return e
+	})
 
-	{
-		var err error
-		state.RequestCancelInfos, err = getRequestCancelInfoMap(
-			ctx,
-			m.db,
-			m.shardID,
-			domainID,
-			wfID,
-			runID,
-			m.parser)
-		if err != nil {
-			return nil, err
-		}
-	}
+	g.Go(func() (e error) {
+		defer recoverPanic(&e)
+		timerInfos, e = getTimerInfoMap(
+			ctx, m.db, m.shardID, domainID, wfID, runID, m.parser)
+		return e
+	})
 
-	{
-		var err error
-		state.SignalInfos, err = getSignalInfoMap(
-			ctx,
-			m.db,
-			m.shardID,
-			domainID,
-			wfID,
-			runID,
-			m.parser)
-		if err != nil {
-			return nil, err
-		}
-	}
+	g.Go(func() (e error) {
+		defer recoverPanic(&e)
+		childExecutionInfos, e = getChildExecutionInfoMap(
+			ctx, m.db, m.shardID, domainID, wfID, runID, m.parser)
+		return e
+	})
 
-	{
-		var err error
-		state.BufferedEvents, err = getBufferedEvents(
-			ctx,
-			m.db,
-			m.shardID,
-			domainID,
-			wfID,
-			runID)
-		if err != nil {
-			return nil, err
+	g.Go(func() (e error) {
+		defer recoverPanic(&e)
+		requestCancelInfos, e = getRequestCancelInfoMap(
+			ctx, m.db, m.shardID, domainID, wfID, runID, m.parser)
+		return e
+	})
+
+	g.Go(func() (e error) {
+		defer recoverPanic(&e)
+		signalInfos, e = getSignalInfoMap(
+			ctx, m.db, m.shardID, domainID, wfID, runID, m.parser)
+		return e
+	})
+
+	g.Go(func() (e error) {
+		defer recoverPanic(&e)
+		bufferedEvents, e = getBufferedEvents(
+			ctx, m.db, m.shardID, domainID, wfID, runID)
+		return e
+	})
+
+	g.Go(func() (e error) {
+		defer recoverPanic(&e)
+		signalsRequested, e = getSignalsRequested(
+			ctx, m.db, m.shardID, domainID, wfID, runID)
+		return e
+	})
+
+	err := g.Wait()
+	if executionsError != nil {
+		return nil, executionsError
+	}
+	if err != nil {
+		return nil, &types.InternalServiceError{
+			Message: fmt.Sprintf("GetWorkflowExecution: failed. Error: %v", err),
 		}
 	}
 
-	{
-		var err error
-		state.SignalRequestedIDs, err = getSignalsRequested(
-			ctx,
-			m.db,
-			m.shardID,
-			domainID,
-			wfID,
-			runID)
-		if err != nil {
-			return nil, err
+	state, err := m.populateWorkflowMutableState(executions[0])
+	if err != nil {
+		return nil, &types.InternalServiceError{
+			Message: fmt.Sprintf("GetWorkflowExecution: failed. Error: %v", err),
```

Comment:
it's probably worth making these two error messages different, so we can tell them apart.  though the causes are quite possibly different between the two.

Reply:
just merged them, I think the impact of separating them should be pretty low